optimistic_lock.hpp

```diff
@@ -312,6 +394,21 @@ class [[nodiscard]] optimistic_lock final {
     }
   }
 
+  // Return a read_critical_section for this optimistic_lock using a
+  // version_tag_type which had been obtained previously.  The use case
+  // for this is to fix up the optimistic_lock when a version_tag_type is
+  // read from the stack for an OLC itertor.  It bumps the read lock
+  // count to make the code happy but does not do any spin waits or
```

Comment:
The `inc_read_lock_count` call should be refactored to a RCS-creating factory method in `optimistic_lock`, removing the need for this comment and cleaning up usage. Not necessary to do now, just please add a TODO note

Reply:
Added TODO